sys/compat/freebsd64/freebsd64_misc.c

```diff
@@ -636,7 +636,7 @@ freebsd64_copyout_strings(struct image_params *imgp, uintcap_t *stack_base)
 	    rounded_stack_vaddr,
 	    CHERI_REPRESENTABLE_LENGTH(ssiz + stack_offset), stack_offset);
 	destp = cheri_setaddress(destp, p->p_sysent->sv_psstrings);
-	arginfo = cheri_setbounds(destp, sizeof(*arginfo));
+	arginfo = cheri_setbounds((void * __capability)destp, sizeof(*arginfo));
```

Comment:
I would probably have put the cast on the "outside" instead of the inside, but perhaps it doesn't matter.  In normal C I tend to put the cast on the outside of an expression unless I need the cast "inside" to do pointer arithmetic.